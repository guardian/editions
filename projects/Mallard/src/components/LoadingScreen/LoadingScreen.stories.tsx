import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { LoadingScreen } from './LoadingScreen';

storiesOf('Loading Screen', module).add('default', () => <LoadingScreen />);
