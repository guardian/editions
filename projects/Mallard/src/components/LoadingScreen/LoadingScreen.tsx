import React from 'react';
import { Dimensions, StyleSheet, View } from 'react-native';
import { Spinner } from '../../components/Spinner/Spinner';

const styles = StyleSheet.create({
	loadingScreen: {
		height: Dimensions.get('window').height,
		width: Dimensions.get('window').width,
		backgroundColor: 'white',
		alignItems: 'center',
		justifyContent: 'center',
	},
});

const LoadingScreen = () => (
	<View style={styles.loadingScreen}>
		<Spinner />
	</View>
);

export { LoadingScreen };
