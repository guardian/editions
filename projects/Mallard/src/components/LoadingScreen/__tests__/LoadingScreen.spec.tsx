import React from 'react';
import TestRenderer from 'react-test-renderer';
import { LoadingScreen } from 'src/components/LoadingScreen/LoadingScreen';

jest.mock('src/components/Spinner/Spinner', () => ({
	Spinner: () => '<Spinner />',
}));

describe('LoadingScreen', () => {
	it('should show a LoadingScreen with default styling', () => {
		const component = TestRenderer.create(<LoadingScreen />).toJSON();
		expect(component).toMatchSnapshot();
	});
});
