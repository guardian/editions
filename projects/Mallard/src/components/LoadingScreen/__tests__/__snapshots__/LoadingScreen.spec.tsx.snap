// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen should show a LoadingScreen with default styling 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  &lt;Spinner /&gt;
</View>
`;
